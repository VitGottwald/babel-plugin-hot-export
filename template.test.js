const template = require("babel-template");
const generate = require("babel-generator").default;
const t = require("babel-types");

it("replaces identifiers in a template", () => {
  const buildRequire = template(`
    var IMPORT_NAME = require(SOURCE);
  `);

  const ast = buildRequire({
    IMPORT_NAME: t.identifier("myModule"),
    SOURCE: t.stringLiteral("my-module")
  });

  expect(generate(ast).code).toMatchSnapshot();
});
