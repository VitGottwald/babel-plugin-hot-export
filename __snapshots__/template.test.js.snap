// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`replaces identifiers in a template 1`] = `"var myModule = require(\\"my-module\\");"`;
